### ISSUE TITLE
Intermittently missing activities when calling GetActivityHistory

### ISSUE THREAD

**Issue by vthornheart-bng** [Bungie-net/api#904]
From @CrunchyWizard47:

"I've got a troublesome activity where I've got three clan members participating in an Iron Banner Control on 3/28 (2019-03-29T02:41:48Z) with an instanceId of 3805062447. It normally doesn't show in the activity list of any of the clan members, but it's in guardian.gg (that's how I found it) and I've had it pop up once or twice (out of about 40 attempts) when hitting https://stats.bungie.net/Platform/Destiny2/1/Account/4611686018443078628/Character/2305843009267861329/Stats/Activities/?count=200&mode=0&page=0 but mostly it doesn't appear...
We require that all clan members play in at least one fireteam with other clan members every calendar month or they get kicked. This member was kicked because the activity didn't show. He protested and that's how I found this. This activity normally doesn't show when I run the activity list of any of the three members: (membershipId/characterId: 4611686018443078628/2305843009267861329, 4611686018461296896/2305843009266719951, 4611686018463061258/2305843009268766526)."

"yes, my script iterates through all activities that are returned for a player and then asks for the PGCR for those which it doesn't already have in my database. My experience using the URL in a browser is that I've seen the instanceId show up only once over around 30 tries (while I don't depend on the activities showing up in chronological order, I find they do and for the URL above, the activities go back to 2018). I understand it's probably caching most of those (and I don't have access to any of those accounts to 'break' the cache) but I've run my script many times (maybe 20?) over the past week and it hasn't captured that instanceId yet. And this is an activity in which three of my members took part, so it should have popped up three times each time I've run it."

---

Has anyone else run into this, where you occasionally get an activity back but most of the time it is missing?

I haven't had the chance to investigate this yet, but if anyone has more info they'd like to add of their own experiences, there's a potential for it to be helpful when I do get around to investigating it!

**Comment by CrunchyWizard47** [comment]
Found another one. The common membershipId/characterId with the above: 4611686018461296896/2305843009266719951, instanceId: 3860533287. This time, only two clan members in the fireteam but the activity doesn't show for either.

**Comment by vthornheart-bng** [comment]
Finally got some time to take a look at these... I'm seeing what feels like different potential root causes for each of these, but I'll need some further input from you on them.

---

For the first one, 3805062447, I'm seeing the following players in it:

CharacterId	AccountId	ClanId
2305843009262051967	4611686018446168803	64755
2305843009262957653	4611686018429759058	2263748
2305843009263481887	4611686018433869181	700125
2305843009265055451	4611686018433989941	3721587
2305843009266400139	4611686018434561348	281304
2305843009266719951	4611686018461296896	495517
2305843009267081704	4611686018453897679	495517
2305843009267861329	4611686018443078628	495517
2305843009268766526	4611686018463061258	495517
2305843009293627176	4611686018470387358	2912579
2305843009319574191	4611686018472862256	1632013
2305843009321967961	4611686018473356778	60441

I hit the GetActivityHistory endpoint a few times at the URL you gave, and every time I got that activity back.  Is it possible that there's some kind of issue with the parsing of the results once you receive them?  Can you save an example of the response you're getting when you *don't* see it (and if possible, if you could include the value of your bungled cookie?  That cookie provides a "session" ID that we can use to see if we hit any unexpected exception/error during processing on our side)

---

Hmm, when I look at the raw data for instance 3860533287, I see the following players - the first of whom has no useful information because they left just as the activity started.  They all seem to be part of different clans:

CharacterId	AccountId	ClanId
2305843009262443293	4611686018433382146	NULL
2305843009266246530	4611686018443627331	629399
2305843009266719951	4611686018461296896	495517
2305843009366464888	4611686018462975233	278058

Does this match what you are seeing?  Are you certain that they were playing this specific instance ID with other clan members?  For this particular activity, it looks like he didn't: at least not according to our data, which would explain why you wouldn't see it if you queried for other members.

---

This second one, 3834776188, shows the following characters, which in this case does seem like it was played by the same clan members:

CharacterId	AccountId	ClanId
2305843009261581158	4611686018442915377	495517
2305843009263098017	4611686018430411101	495517
2305843009268176255	4611686018433914220	495517

That one seems to be coming through with them when I hit the PGCR directly, and also when I hit the GetActivityHistory endpoint for any one of the users - but it is several pages in.  How frequently are you finding it is missing in your results in this situation?  Are these the specific accounts you were expecting to see playing together?

**Comment by CrunchyWizard47** [comment]
Sorry for the delay (been busy with my paying gig :) ) - I'm digging more deeply into this now and I found an issue in my script that parses the json. It was odd that the activity wasn't showing consistently when I would query the API using Chrome (that's when I posted), but now I'm seeing the activities consistently so I need to fix the script. I'll update if I see the behavior again (along with the cookies).
